omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController2_0Test.java

```diff
@@ -94,6 +99,58 @@ public void shouldGetRefByUuid() {
 	public void shouldGetFullByUuid() {
 	}
 	
+	@Test
+	public void createAttachment_shouldCreateANewAttachment() throws Exception {
+		SimpleObject attachment = new SimpleObject();
+		attachment.add("uuid", obs.getUuid());
+		attachment.add("comment", obs.getComment());
+		attachment.add("complexData", obs.getComplexData());
+		
+		String json = new ObjectMapper().writeValueAsString(attachment);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		Util.log("Created attachment", attachment);
+		
+		// Check existence in database
+		String uuid = (String) attachment.get("uuid");
+		Assert.assertNotNull("uuid", obs.getUuid());
+		Assert.assertNotNull("comment", obs.getComment());
+		Assert.assertEquals("complexData", obs.getComplexData(), null);
+		
+	}
+	
+	@Test
+	public void getAttachment_shouldGetADefaultRepresentationOfAttachment() throws Exception {
+		
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		Assert.assertNotNull(result);
+		Util.log("Attachment fetched (default)", result);
+		Assert.assertEquals(getUuid(), result.get("uuid"));
+	}
+	
+	@Test
+	public void updateAttachment_shouldChangeAPropertyOnAttachment() throws Exception {
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("complexData", "update complextData");
+		attributes.add("deathDate", "Updated deathDate");
+		attributes.add("prefferedName", "Updated   prefferedName");
+		attributes.add("gender", "updated gender");
+		attributes.add("preferredAddress", "updated preferredAddress");
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		
+		String editedAttachment = obs.getUuid();
+		Assert.assertNotEquals("Updated complexData ", editedAttachment.equalsIgnoreCase(editedAttachment));
```

Comment:
@sherrif10   , same here , you need to check the deserialized response , not the OBS object you posted

Reply:
@mozzy11  ok no problem